lib/src/bindings/signatures.dart

```diff
@@ -55,3 +55,70 @@ typedef obx_box_put_native_t = Int32 Function(Pointer<Void> box, Uint64 id, Poin
+
+typedef obx_qb_cond_operator_2_native_t<P> = Int32 Function(Pointer<Void> builder, Uint32 property_id, P v1, P v2);
+typedef obx_qb_cond_operator_2_dart_t<P>   = int Function(Pointer<Void> builder, int property_id, P v1, P v2);
+
+typedef obx_qb_cond_operator_in_native_t<P> = Int32 Function(Pointer<Void> builder, Uint32 property_id, Pointer<P> values, Uint64 count);
+typedef obx_qb_cond_operator_in_dart_t<P>   = int Function(Pointer<Void> builder, int property_id, Pointer<P> values, int count);
+
+typedef obx_qb_join_op_native_t = Int32 Function(Pointer<Void> builder, Pointer<Int32> cond_array, Uint64 count);
+typedef obx_qb_join_op_dart_t   = int Function(Pointer<Void> builder, Pointer<Int32> cond_array, int count);
+
+typedef obx_qb_cond_string_op_1_native_t = Int32 Function(Pointer<Void> builder, Uint32 property_id, Pointer<Uint8> value, Int8 case_sensitive);
+typedef obx_qb_cond_string_op_1_dart_t   = int Function(Pointer<Void> builder, int property_id, Pointer<Uint8> value, int case_sensitive);
+
+typedef obx_qb_string_lt_gt_op_native_t = Int32 Function(Pointer<Void> builder, Uint32 property_id, Pointer<Uint8> value, Int8 case_sensitive, Int8 with_equal);
+typedef obx_qb_string_lt_gt_op_dart_t   = int Function(Pointer<Void> builder, int property_id, Pointer<Uint8> value, int case_sensitive, int with_equal);
+
+typedef obx_qb_string_in_native_t = Int32 Function(Pointer<Void> builder, Uint32 property_id, Pointer<Pointer<Uint8>> value, Uint64 count, Int8 case_sensitive);
+typedef obx_qb_string_in_dart_t   = int Function(Pointer<Void> builder, int property_id, Pointer<Pointer<Uint8>> value, int count, int case_sensitive);
+
+typedef obx_qb_bytes_eq_native_t = Int32 Function(Pointer<Void> builder, Uint32 property_id, Pointer<Void> value, Uint64 size);
+typedef obx_qb_bytes_eq_dart_t   = int Function(Pointer<Void> builder, int property_id, Pointer<Void> value, int size);
+
+typedef obx_qb_bytes_lt_gt_native_t = Int32 Function(Pointer<Void> builder, Uint32 property_id, Pointer<Void> value, Uint64 size, Int8 with_equal);
+typedef obx_qb_bytes_lt_gt_dart_t   = int Function(Pointer<Void> builder, int property_id, Pointer<Void> value, int size, int with_equal);
+
+typedef obx_qb_param_alias_native_t = Int32 Function(Pointer<Void> builder, Pointer<Uint8> alias);
+typedef obx_qb_param_alias_dart_t   = int Function(Pointer<Void> builder, Pointer<Uint8> alias);
+
+typedef obx_qb_order_native_t = Int32 Function(Pointer<Void> builder, Uint32 property_id, Uint32 flags);
+typedef obx_qb_order_dart_t   = int Function(Pointer<Void> builder, int property_id, int flags);
+
+// query
+
+typedef obx_query_t = Pointer<Void> Function(Pointer<Void> builder);
+
+typedef obx_query_close_native_t = Int32 Function(Pointer<Void> query);
+typedef obx_query_close_dart_t =   int Function(Pointer<Void> query);
+
+typedef obx_query_find_t<T> = Pointer<Uint64> Function(Pointer<Void> query, T offset, T limit);
```

Comment:
this still doesn't seem right, I'd have expected something more similar to box_get_all (`	typedef obx_box_get_all_native_t = Pointer<Uint64> Function(Pointer<Void> box);`) and a separate method for find and find_ids (they have different return types

```suggestion
typedef obx_query_find_t = Pointer<Void> Function(Pointer<Void> query, Uint64 offset, Uint64 limit);
typedef obx_query_find_ids_t = Pointer<Uint64> Function(Pointer<Void> query, Uint64 offset, Uint64 limit);
```

but maybe I'm missing something

Reply:
I didn't feel like repeating myself (DRY), if the generic function can switch out the `int` for `Uint64`.